core/src/test/java/org/apache/iceberg/TestUpdatePartitionSpec.java

```diff
@@ -614,6 +605,65 @@ public void testRemoveAndAddMultiTimes() {
     V2Assert.assertEquals("Should match expected spec", v2Expected, updated);
   }
 
+  @Test
+  public void testRemoveAndUpdateWithIdentity() {
+    PartitionSpec expected = PartitionSpec.builderFor(SCHEMA)
+         .identity("ts")
+         .build();
+    PartitionSpec updated = new BaseUpdatePartitionSpec(formatVersion, expected)
+         .removeField("ts")
+         .addField("ts")
+         .apply();
+
+    if (formatVersion == 1) {
+      Assert.assertEquals("Should match expected spec field size", 2, updated.fields().size());
+      Assert.assertEquals("Should match expected field name", "ts_1000",
+              updated.fields().get(0).name());
+      Assert.assertEquals("Should match expected field name", "ts",
+              updated.fields().get(1).name());
+      Assert.assertEquals("Should match expected field transform", "void",
+              updated.fields().get(0).transform().toString());
+      Assert.assertEquals("Should match expected field transform", "identity",
+              updated.fields().get(1).transform().toString());
+    } else {
+      Assert.assertEquals("Should match expected spec field size", 1, updated.fields().size());
+      Assert.assertEquals("Should match expected field name", "ts",
+              updated.fields().get(0).name());
+      Assert.assertEquals("Should match expected field transform", "identity",
+              updated.fields().get(0).transform().toString());
+    }
+  }
+
+  @Test
+  public void testRemoveAndUpdateWithDifferentTransformation() {
+    PartitionSpec expected = PartitionSpec.builderFor(SCHEMA)
+        .month("ts", "ts_transformed")
+        .build();
+    PartitionSpec updated = new BaseUpdatePartitionSpec(formatVersion, expected)
+        .removeField("ts_transformed")
+        .addField("ts_transformed", day("ts"))
+        .apply();
+
+    if (formatVersion == 1) {
+      Assert.assertEquals("Should match expected spec field size", 2, updated.fields().size());
```

Comment:
You can also use `V2Assert` and `V1Assert` to avoid checking the format version explicitly.

Reply:
Hi, @rdblue thanks for the review, I think the problem with V1Assert/V2Assert here is that we still need to check the length of updated.fields for different versions of format

maybe checking version format is the most straightforward way like [here](https://github.com/apache/iceberg/blob/57bdf0ba61b10944f5fa6045cc32e321e90eb746/core/src/test/java/org/apache/iceberg/TestUpdatePartitionSpec.java#L584)